src/operator/src/statement/copy_database.rs

```diff
@@ -0,0 +1,250 @@
+use crate::error::{CatalogSnafu, InvalidCopyParameterSnafu};
+use crate::statement::StatementExecutor;
+
+pub(crate) const COPY_DATABASE_TIME_START_KEY: &str = "start_time";
+pub(crate) const COPY_DATABASE_TIME_END_KEY: &str = "end_time";
+pub(crate) const CONTINUE_ON_ERROR_KEY: &str = "continue_on_error";
+
+impl StatementExecutor {
+    #[tracing::instrument(skip_all)]
+    pub(crate) async fn copy_database_to(&self, req: CopyDatabaseRequest) -> error::Result<Output> {
+        // location must end with / so that every table is exported to a file.
+        ensure!(
+            req.location.ends_with('/'),
+            InvalidCopyParameterSnafu {
+                key: "location",
+                value: req.location,
+            }
+        );
+
+        info!(
+            "Copy database {}.{} to dir: {}, time: {:?}",
+            req.catalog_name, req.schema_name, req.location, req.time_range
+        );
+        let table_names = self
+            .catalog_manager
+            .table_names(&req.catalog_name, &req.schema_name)
+            .await
+            .context(CatalogSnafu)?;
+
+        let suffix = Format::try_from(&req.with)
+            .context(error::ParseFileFormatSnafu)?
+            .suffix();
+
+        let mut exported_rows = 0;
+        for table_name in table_names {
+            // TODO(hl): remove this hardcode once we've removed numbers table.
+            if table_name == "numbers" {
+                continue;
+            }
+            let mut table_file = req.location.clone();
+            table_file.push_str(&table_name);
+            table_file.push_str(suffix);
+            info!(
+                "Copy table: {}.{}.{} to {}",
+                req.catalog_name, req.schema_name, table_name, table_file
+            );
+
+            let exported = self
+                .copy_table_to(
+                    CopyTableRequest {
+                        catalog_name: req.catalog_name.clone(),
+                        schema_name: req.schema_name.clone(),
+                        table_name,
+                        location: table_file,
+                        with: req.with.clone(),
+                        connection: req.connection.clone(),
+                        pattern: None,
+                        direction: CopyDirection::Export,
+                        timestamp_range: req.time_range,
+                    },
+                    QueryContextBuilder::default().build(),
+                )
+                .await?;
+            exported_rows += exported;
+        }
+        Ok(Output::AffectedRows(exported_rows))
+    }
+
+    /// Imports data to database from a given location and returns total rows imported.
+    #[tracing::instrument(skip_all)]
+    pub(crate) async fn copy_database_from(
+        &self,
+        req: CopyDatabaseRequest,
+        ctx: QueryContextRef,
+    ) -> error::Result<Output> {
+        // location must end with /
+        ensure!(
+            req.location.ends_with('/'),
```

Comment:
The same suggestion as above, make the error message more friendly in this case.

Reply:
Resolved https://github.com/GreptimeTeam/greptimedb/commit/1da7d2e6b7e02f972b92d52e23d821af060a1361#diff-b3f149877e58dcda21272ac161833e8417f156d325d0d74465f7fc5f0f70a8a5R47